app/views/assessments/viewGradesheet.html.erb

```diff
@@ -238,6 +246,61 @@ Interaction between gradesheet and annotations
 <span id="interaction_badge" class="badge" data-badge-caption="Feature Information"></span></a>
 </div>
 
+<script type="text/javascript">
+    var loadFunc = (elem) => {
+        console.log("Value saved is: "+ localStorage.getItem("savedValue"));
+        if(localStorage.getItem("savedValue")){ //checks if value is saved or not
+            elem.checked = localStorage.getItem("savedValue");
+        }
+    }
+    var clickFunc = (elem) => {
+        localStorage.setItem("savedValue", elem.checked);
+    }
+</script>
+
+<div class="switch">
+  <label>
+    <b>Show students with completed submissions</b>
+    <% if params[:viewall] == "true" %>
+      <input type="checkbox" onclick="clickFunc(this)" id="view_all_checkbox" checked>
+    <% else %>
+      <input type="checkbox" onclick="clickFunc(this)" id="view_all_checkbox">
+    <% end %>
+    <span class="lever"></span>
+    <b>Show all students </b>
+  </label>
+</div>
+<label for="switch">
+  All students includes those with missing submissions
+</label>
+
+<script defer>
+    document.addEventListener('DOMContentLoaded', function() {
+        var checkBox = document.getElementById('view_all_checkbox');
+
+        function updateURL() {
```

Comment:
why not just redirect to `viewGradesheet` with the param set, instead of doing all of this url manipulation and reloading?